File: src/main/java/snake/difficulty/Advanced.java
package snake.difficulty;

public class Advanced extends Difficulty {
    public Advanced() {
        this.delay = 100;
    }

    @Override
    public void changeDelay(int foodConsumed) {
        if (delay > 30 && foodConsumed % 4 == 0) {
            delay -= 20;
        }
    }
}

File: docs/scripts/difficulty/Advanced.js
import Difficulty from "./Difficulty.js";

export default class Advanced extends Difficulty {
    constructor() {
        super(100);
    }
    changeDelay(foodConsumed) {
        if (this.delay > 30 && foodConsumed % 4 === 0) {
            this.delay -= 20;
        }
    }
}
